ArduCopter/mode_smart_rtl.cpp

```diff
@@ -86,21 +86,23 @@ void ModeSmartRTL::path_follow_run()
 
     // if we are close to current target point, switch the next point to be our target.
     if (wp_nav->reached_wp_destination()) {
-        Vector3f next_point;
-        if (g2.smart_rtl.pop_point(next_point)) {
-            bool fast_waypoint = true;
-            if (g2.smart_rtl.get_num_points() == 0) {
-                // this is the very last point, add 2m to the target alt and move to pre-land state
-                next_point.z -= 2.0f;
+        if (g2.smart_rtl.get_num_points() != 0) {
```

Comment:
Couldn't this be done by changing the else clause on line 102 instead?

i.e. check to see *why* the `get_num_points` failed, and if it is because the list is empty then we can go to the next state.

BTW, what *is* the race condition here?  Can we attempt to detect the state of the thread and avoid the race condition on `pop_point`?